[Donny Suriyanto] iP

DonnyS57 opened the pull request:
## Alexa

> "Your mind is for having ideas, not holding them." - David Allen [(source)](https://dansilvestre.com/productivity-quotes) 

Alexa frees your mind of having to remember things you need to do. It's, 
 * text-based
 * easy to learn
 * ~~FAST~~ *SUPER FAST* to use

All you need to do is,
1. Download it from [here](https://github.com/DonnyS57/ip).
1. Double-click it.
1. Add your tasks.
1. Let it manage your tasks for you :wink:

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

## 

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

DonnyS57 committed:
Finish Level 1.

DonnyS57 committed:
Finish Level 2

DonnyS57 committed:
Finish Level 3

DonnyS57 committed:
Finish Level 3

DonnyS57 committed:
Finish Level 4

DonnyS57 committed:
Finish A-TextUiTexting

DonnyS57 committed:
Finish Level 5

DonnyS57 committed:
Finish Level 6

DonnyS57 committed:
Finish Level 7

DonnyS57 committed:
Finish Level 8

DonnyS57 committed:
Merge branch 'branch-Level-8'

DonnyS57 committed:
Push Texts

DonnyS57 committed:
Finish A-MoreOOP

DonnyS57 committed:
Finish A-Packages

solarrabbit99 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+    private TaskList tasks;
+    private Ui ui;
+    private static Parser parser = new Parser();
+
```
I believe it was mentioned in one of the later tasks that at least half of the public classes and methods are supposed to have a header comment/javadoc.

solarrabbit99 commented on src/main/java/duke/Time.java:
```diff
@@ -0,0 +1,28 @@
+        SimpleDateFormat sdfrmt = new SimpleDateFormat("dd/MM/yyyy");
+        sdfrmt.setLenient(false);
+
+        try
```
Is this considered Egyptian-styled brackets? I think it safer as per following:
```java
try {
} catch (Exception e) { // Narrow the catch statement
}
```

solarrabbit99 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,16 @@
+
+public class Deadline extends duke.Task {
+
+    private String by;
```
I would have preferred names like `deadline` or `dateString` that are more descriptive. See if you agree with that.

solarrabbit99 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,92 @@
+                        break;
+                    default:
+                        break;
+                }
```
I believe there are supposed to have no indentations for case clauses.

solarrabbit99 commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,14 @@
+package duke;
+import duke.Task;
+
+public class Todo extends Task {
+
+    public Todo(String description) {
+        super(description);
+    }
+
+    @Override
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+}
```
I like the simplicity of the class! Good practice of inheritance!

solarrabbit99 reviewed (commented):
Good work!

marcusc55 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+        }
+    }
+
+    public static void main(String[] args) {
```
Should these methods have JavaDoc comments?

marcusc55 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,18 @@
+package duke;
+import java.time.format.DateTimeFormatter;
+
+public class Event extends duke.Task {
```
Would it be better to change `extends duke.Task` to `extends Task`, since both classes are in the same package?

marcusc55 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,18 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (at:" + date + ")";
```
Would it be better to change the `[E]` string to a named constant? I noticed this for the other classes that extend from Task as well.

marcusc55 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    public static void print(String text) {
+        System.out.println("=======================================");
```
Would it be better to extract the strings in this class (e.g. `==========`, `Congratulations on finishing this task!`) into named constants?

marcusc55 reviewed (commented):
Your code is clear and understandable, other than some small comments. LGTM!

DonnyS57 committed:
Finish A-JUnit

DonnyS57 committed:
Finish A-Jar

DonnyS57 committed:
Added Javadocs

DonnyS57 committed:
Adhere Coding Standard

DonnyS57 committed:
Finish Level 9

DonnyS57 committed:
Merge branch 'branch-Level-9' into branch-A-JavaDoc

DonnyS57 committed:
Merge branch 'branch-A-JavaDoc' into branch-A-CodingStandard

DonnyS57 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

DonnyS57 committed:
Fixed Coding Standard Issue

DonnyS57 committed:
Add JavaFX

DonnyS57 committed:
Add Assert functions

Added 2 assert functions to the program. 1 was added to the Duke class
to ensure that when the function goes through all of the if-else
statements, it is because no valid input has been inserted by the user.

The second assert function was added to the Storage class. It is used to
detect if there were any invalid task type that somehow managed to get
into the saved text file.

DonnyS57 committed:
Fix Code Quality

Most of the code are already in the code quality standard. However,
there are still some magic numbers being used in certain classes such as
Parser. As such, some magic numbers are changed to a constant for better
readability.

DonnyS57 committed:
Merge pull request #2 from DonnyS57/branch-A-Assertions

Add Assert functions

DonnyS57 committed:
Merge branch 'master' into branch-A-CodeQuality

DonnyS57 committed:
Merge pull request #3 from DonnyS57/branch-A-CodeQuality

Fix Code Quality

DonnyS57 committed:
Add GUI to Alexa

DonnyS57 committed:
Merge branch 'branch-Level-10'

DonnyS57 committed:
Fix Checkstyle Issues

DonnyS57 committed:
Fix Issue with Storage

DonnyS57 committed:
Add C-FriendlierSyntax functionality

DonnyS57 committed:
Add Ui.png

DonnyS57 committed:
Update ReadME.md

DonnyS57 committed:
Update README.md

DonnyS57 committed:
Merge branch 'master' of https://github.com/DonnyS57/ip

DonnyS57 committed:
Change Title

DonnyS57 committed:
Delete Old Ui.png

DonnyS57 committed:
Update README.md

DonnyS57 committed:
Update README.md

DonnyS57 committed:
Set theme jekyll-theme-cayman
